### PR TITLE
Read EngDiff save location at runtime

### LINKED ISSUE #33877: Save location not being used in all situations in Engineering Diffraction UI
a) clear down all EngineeringDiffraction2 settings in MantidWorkbench.ini fle
b) start up workbench
c) open up Engineering Diffraction UI
d) Create New Calibration on Calibration tab using run 305738

Some .nxs and .prm files should be saved into C:\\Users\\iao46725\\Engineering_Mantid\Calibration\ as indicated by the "Save Location" setting within the settings for this UI (cog button on bottom left of engineering diffraction UI)

Although the default save location has been set up as something like C:\Users\iao46725\Engineering_Mantid\ it doesn't get used to save the files. Some of the files are instead saved into a Calibration subdirectory of whatever the current working directory happens to be - this can vary. And some don't get written at all. The following error is also produced:

image

If you shut down workbench and start it up again it works OK.

Initial investigation suggests the result of output_settings.get_output_path() is a blank string on first start up of workbench (called in qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/calibration/model.py). The default doesn't seem to kick in properly at first.

### PR DESCRIPTION
**Pull request by DanielMurphy22**
**Description of work.**

The default value for the `save_dir` parameter in the calibration model was set at compile time (when the 
function was defined) not at runtime. Now instead of setting the default parameter to a function return value:
`def create_new_calibration(... save_dir=output_settings.get_output_path()`
the default parameter is set to None, so the `output_settings.get_output_path()` function runs at runtime.

Also fixed this bug in the focus model.

Had to read this article for inspiration: https://medium.com/techtofreedom/mind-the-default-arguments-in-python-functions-8b00d6f532e3

**To test:**

- Rename your mantidworkbench.ini file, see the steps for [**Config Files** in the relevant section for Windows, Linux and MacOS ](https://developer.mantidproject.org/UserSupport.html).
- Open MantidWorkbench
- Open the Engineering Diffraction interface
- Create a new calibration with the file /archive/ndxenginx/Instrument/data/cycle_19_1/ENGINX00305738.nxs (path on the archive may be different on Windows)
- The calibration should complete successfully, with no errors and should save calibration files somewhere in an `Engineering_Mantid` folder. Check where this is by looking in the Engineering Diffraction settings (gear icon bottom left of interface) and check some files are added to this directory in a file browser.

Fixes #33877

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/calibration/model.py
Comment (AndreiSavici) [inline qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/calibration/model.py:18]: Have you tried to run this function with `savedir='/tmp/'` argument? On my toy model I get `TypeError: 'str' object is not callable`

**Review by robertapplin — CHANGES_REQUESTED** [review]
Nice job finding the problem, its now working as expected. Before I approve, I just wanted to ask if it would be easy/possible to add a test for this? If not then no worries

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/calibration/test/test_calib_model.py
Comment (robertapplin) [inline qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/calibration/test/test_calib_model.py:28]: Is this meant to be here still?

**Review by robertapplin — CHANGES_REQUESTED** [review]
Thanks for the hard work adding the tests, they look great! Just had one small question and then I'll approve

File: qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/calibration/test/test_calib_model.py
Comment (DanielMurphy22) [reply qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/calibration/test/test_calib_model.py]: Good spot!

**Review by robertapplin — APPROVED** [review]
Nice work, looks good to go 👍

**Pull request merged**
